doc/manual/spec

```diff
@@ -214,24 +214,67 @@ can express this as
 	BuildConflicts: gcc <= 2.7.2.1
 \endverbatim
 
+\section comments Comments
+
+Comments in spec file have # at the start of the line.
+\verbatim
+	# this is a comment
+\endverbatim
+Macros are expanded in comments, thus if a macro is in comment use two percent
+characters before it:
+\verbatim
+	# make unversioned %%__python an error unless explicitly overridden
+\endverbatim
+
 \section conditionals Conditionals
 
-RPM's spec file format allows conditional blocks of code to be used dependent
-on various properties such as architecture (%ifarch / %ifnarch), operating system
-(%ifos / %ifnos), or any other conditional expression (%if). Conditionals may be
-nested within other conditionals:
+RPM's spec file format allows conditional blocks of code to be used
+depending on various properties such as architecture (%ifarch /%ifnarch),
+operating system (%ifos / %ifnos), or a conditional expression (%if).
 
+%ifarch is generally used for building RPM packages for multiple
+platforms like:
 \verbatim
-	%ifarch ppc64 ppc x86_64
-	...
-	%if 0%{?ver} && 0%{?ver} >= 100
-	...
-	%else
-	...
-	%endif
+	%ifarch %{power64} s390x x86_64 ia64
+	%global architectures 1
 	%endif
 \endverbatim
 
-Conditionals are not macros, thus they can be used in spec files only.
+%ifos is used to control RPM's spec file processing according to the
+build system's operating system.
+
+%if can be used for various purposes. The test can be evaluated based on
+the existence of a macro, like:
+\verbatim
+	%if %{defined with_foo} && %{undefined with_bar}
+\endverbatim
+string comparison:
+\verbatim
+	%if %{optimize_flags} != "none"
+\endverbatim
+or a mathematical statement:
+\verbatim
+	%if 0%{?fedora} > 10 || 0%{?rhel} > 7
+\endverbatim
+Generally, a mathematical statement allows to use logical operators
+&&, ||, !, relational operators !=, ==, <, > , <=, >= and parentheses.
+
+The conditional blocks end by %endif. Inside the conditional block %elif,
+%elifarch, %elifos or %else can be optionally used. Conditionals %endif and
+%else should not be followed by any text. Conditionals may be nested within
+other conditionals.
+
+Conditionals are not macros. It has an interesting consequence (similar as
+the fact that macros get expanded in comments). If a conditional is a part
+of a definition like
+\verbatim
+	%define macro_with_if \
+	%if 0  \
+	%global output XXX\
+	%endif
+\endverbatim
+the expansion (including %global) occurs before evaluating of the
+conditionals. Thus after the expansion of %macro_with_if macro %output
+is defined to XXX.
```

Comment:
I don't think this last block belongs here, at least in this detail. Dark corners are often best left undocumented to allow changing if a saner behavior becomes feasible later. I'd condense that into something like:

%if-conditionals are not macros, and are unlikely to yield expected results if used in them.